### PR TITLE
Only support iSCSI when storaged can list sessions

### PR DESCRIPTION
**Pull request by mvollmer**
Depending on the patchlevel of libiscsi, storaged might or might not
be able to list iSCSI sessions.

### CODE REVIEW

**Comment by mvollmer** [comment]
This has just the test.  The actual code to detect whether sessions are supported is still missing.

**Comment by mvollmer** [comment]
Hmm, how did check-storage-iscsi manage to pass on rhel-7?

**Comment by mvollmer** [comment]
> Hmm, how did check-storage-iscsi manage to pass on rhel-7?

Race condition, of course.

**Comment by mvollmer** [comment]
I had to procrastinate a bit over how to detect whether storaged actually supports iscsi sessions.... and I don't think there is a good way, unfortunately.  So I added a configure flag...

**Comment by stefwalter** [comment]
Could you explain why the configure flag was the best option? Is there nothing in the DBus API that can tell us about this?

**Comment by mvollmer** [comment]
> Could you explain why the configure flag was the best option? Is there nothing in the DBus API that can tell us about this? 

Correct.  Storaged detects at compile-time whether or not the libiscsi session API is available, and simply omits the creation of session objects when it is not there.

Thus, we can't check the version of storaged.  We could check the symbols inside the libstoraged_iscsi.so module but that is also brittle, of course.

**Comment by stefwalter** [comment]
I'm fine with merging this as a workraound but I think we need to actually help add this capability to storaged. We need to have it be able to tell us this information via its API.

File: pkg/storaged/config.json.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (stefwalter) [inline pkg/storaged/config.json.in]: Can this go into an override.json that's installed, and loaded as part of the package manifest?

**Comment by mvollmer** [comment]
> I'm fine with merging this as a workraound but I think we need to actually help add this capability to storaged. We need to have it be able to tell us this information via its API.

Or storaged could disable all of iSCSI unless it can also support sessions.  Exposing the messy maintenance situation of the iSCSI stack in the storaged API isn't very helpful.

**Comment by mvollmer** [comment]
I have filed https://bugzilla.redhat.com/show_bug.cgi?id=1352454

**Comment by mvollmer** [comment]
Also, https://github.com/storaged-project/storaged/pull/68

File: pkg/storaged/config.json.in (hunk 1 above)
Comment (stefwalter) [inline pkg/storaged/config.json.in]: This file should be removed, no?

File: configure.ac [hunk 2]
```diff
@@ -304,6 +304,15 @@ changequote([,])dnl
 
 AC_CHECK_FUNCS(fdwalk)
 
+# Storaged
+
+AC_ARG_WITH(storaged-iscsi-sessions,
```
Comment (stefwalter) [inline configure.ac]: I would suggest marking this as HACK: with the appropriate https://github.com/storaged-project/storaged/pull/68 link.

**Comment by mvollmer** [comment]
storaged-project/storaged#68 will be in storaged 2.5.3 so I will rework this to disable iSCSI also when `SessionSupport === false`.  (But not when `SessionSupport === undefined`).

**Pull request closed without merging**